sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
```

Comment:
This comment is misleading it should say something along the lines of:

this fails if the DISPLAY environment variable is not set and the previous line did not succeed in changing the backend, for example because import matplotlib.pyplot was imported before trying to set the backend to 'Agg'.